backend/eventloop/cluster_reconciler.go

```diff
@@ -0,0 +1,206 @@
+package eventloop
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-logr/logr"
+	managedgitopsv1alpha1 "github.com/redhat-appstudio/managed-gitops/backend-shared/apis/managed-gitops/v1alpha1"
+	sharedutil "github.com/redhat-appstudio/managed-gitops/backend-shared/util"
+	logutil "github.com/redhat-appstudio/managed-gitops/backend-shared/util/log"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/discovery"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	// Interval in Minutes to reconcile ClusterReconciler.
+	orphanedResourcesCleanUpInterval = 30 * time.Minute
```

Comment:
I'm just pulling numbers out of my butt here :grin:, but what about every 2 hours? Scanning through all the resources of all the namespaces on the cluster is going to be a fairly intensive process, so we probably don't want to do it TOO often.